ecss services and software.

nchronas opened the pull request:
<!---
@huboard:{"custom_state":"archived","order":26,"milestone_order":26}
-->

nchronas committed:
project from cubeMX

Configured with USART2, FAT system on SD and FreeRTOS. It will be used
for a testing platform on a stm32F4 discovery kit.

nchronas committed:
adding testing files

nchronas committed:
s testing

trying to use a sd card

nchronas committed:
working sd, uart2 , No freertos

nchronas committed:
i2c  imu who iam works

nchronas committed:
spi mag working

nchronas committed:
pwm working

nchronas committed:
adc working

nchronas committed:
tc/tm defitions

nchronas committed:
updated struct

nchronas committed:
adding untested logic

nchronas committed:
unit test for hldlc pack

nchronas committed:
unit tested

nchronas committed:
adding services and UT for tc_tm

nchronas committed:
added checking

nchronas committed:
finishing UT

nchronas committed:
adding services

nchronas committed:
fixing and cleaning the code

nchronas committed:
fxing and cleaning code 2

nchronas committed:
code fixing & cleaning 3

nchronas committed:
finishing verification service

nchronas committed:
almost there

nchronas committed:
fixing serial.c

nchronas committed:
one step closer

nchronas committed:
adding hal layer

nchronas committed:
it compiles!

nchronas committed:
fixing main

nchronas committed:
changing baud rate

nchronas committed:
adding pkt memory pool management

nchronas committed:
adding mass storage service

nchronas committed:
changing tc tm pkt to typedef

nchronas committed:
ldata & mass storage service functionality

nchronas committed:
adding functionality and test service

nchronas committed:
finished design

nchronas committed:
converting indentation to spaces

nchronas committed:
add some assert & require, functionality in large data and mass storage

nchronas committed:
finished mass storage, not tested.

nchronas committed:
cleaned code

nchronas committed:
removed pointer for part

nchronas committed:
added more functionality in MS & LD

nchronas committed:
added functionality in ldata service

nchronas committed:
updated ToDo list, finishing ldata, created function management service

nchronas committed:
updated ldata functionality

nchronas committed:
refined ldata service

nchronas committed:
refactoring code

nchronas committed:
added functionality in route_verification for serial communication and refined code

nchronas committed:
fixed OBC etc duplicate definition, fixed assert definition

nchronas committed:
fixed assert

nchronas committed:
added TC_MAX_PKT_SIZE definition for assertion

nchronas committed:
added forgotten assertion true

nchronas committed:
fixed pointer reference & wrong statement

nchronas committed:
added correct assertion definition

nchronas committed:
cleaned code, added definitions

nchronas committed:
changed to correct variable, added extern variable huart

nchronas committed:
fixed pwr ctrl, assertions and definitions

nchronas committed:
fixed route verification, assertions and definitions

nchronas committed:
fixed app id definitions, added packet mode in tc_tm

nchronas committed:
added definition in .h file

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed assertions and wrong statements

nchronas committed:
fixed assertions and declerations

nchronas committed:
removed forgotten assertion

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed half of errors in ldata

nchronas committed:
fixed variables type

nchronas committed:
fixed some minor mistakes in Ldata, added definitions in pkt pool

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed assertions and declerations

nchronas committed:
fixed assertions and declerations

nchronas committed:
removed erroneous stub

nchronas committed:
fixed some errors in ms

nchronas committed:
compiles but no link

nchronas committed:
adding definition in .c for linker issues

nchronas committed:
compiles and links, needs testing

nchronas committed:
cleaned repo, added route for test service

nchronas committed:
reduced warnings, added extern functions.

nchronas committed:
defining more extern functions.

nchronas committed:
migrated route verification in service utilities

nchronas committed:
renamed tc_tm to services

nchronas committed:
made renames for tc_tm header

nchronas committed:
renamed various definitions

mashua committed:
Refactoring return states.

nchronas committed:
cleaned code

nchronas committed:
refactoring structure

nchronas committed:
cleaned repo

nchronas committed:
fixed more warnings

nchronas committed:
minimized warnings

nchronas committed:
removed stack.c and renamed su

nchronas committed:
added missing definition

nchronas committed:
used safer snprintf

nchronas committed:
fixed snprintf argumens

nchronas committed:
moved global declaretion in .c

nchronas committed:
moved global declaretion in .c

nchronas committed:
changed generic type to correct enum

nchronas committed:
added get pkt in report

nchronas committed:
fixed previous declaretion error

nchronas committed:
fixed issues

nchronas committed:
fixed various issues

nchronas committed:
fixed previous errors

nchronas committed:
changed how verification service by adding verification variable in tc tm header

nchronas committed:
fixed typos

nchronas committed:
added check for max files stores

nchronas committed:
fixed previous errors

nchronas committed:
added global counters for file storage sizes and files

nchronas committed:
added event log, WIP, it will not compile

nchronas committed:
removed definitions, added comments

nchronas committed:
commented out event log in order to check compilation

nchronas committed:
added padding for checksum field

nchronas committed:
removed unused definitions. added in unpack, pack

nchronas committed:
reviewed services .c .h

mashua commented on services/services.c:
```diff
@@ -0,0 +1,60 @@
+    return 0;
+}
+
+uint8_t tst_debugging() {
```
Add params as: (uint8_t\* file, uint32_t line) and implement the actual function (stub removal).
Also update the signature in .h file.

nchronas committed:
reviewed till unpack, added definitions

nchronas committed:
finished review of service utilities

nchronas committed:
reviewed verification service

nchronas committed:
reviewed hk

nchronas committed:
finished changes in redefinitions

nchronas committed:
updated hal uart rx in order to output ok when it received a byte.

nchronas committed:
updated test project for cubemx 4.14 and f4 firmware 1.10.

nchronas committed:
made fix for pkt test service in order to work

hldlc frame should be checked, not 100% correct.
needed to change the assert definition in order to work, more
investigation needed.
pkt pool and crt need ** pointer to work.

nchronas committed:
fm led on/off works.

nchronas committed:
added a small python program that outputs bytes to serial. used for temp testing only. the pkts are hardcoded from tc_tm_UT.c

nchronas committed:
added implementation of assert

need to add,
debug port instead of serial,
error result for log book inside of tst_debugging,
make a faster implementation.

nchronas committed:
updated project

nchronas committed:
fixed verification service.

hldlc needs to be checked again

nchronas committed:
added tests for verification

nchronas committed:
fixed hk service and added tests

nchronas committed:
added su load script

nchronas committed:
added init function

nchronas committed:
added functionality in su populate

nchronas committed:
added su functionality

nchronas committed:
finished most of the su functionality

nchronas committed:
added functionality for su mnlp, pending for timekeeping.

nchronas committed:
added event log.

nchronas committed:
various changes for IAR

nchronas committed:
added event log

nchronas committed:
fixed event log, verfied rtc works

nchronas committed:
added tests for reset sources.

nchronas committed:
various project updates.

nchronas committed:
added unique filed id, for assertion

nchronas committed:
removed the pkt pool modes, only extended is used

nchronas committed:
fixed pkt pool mode change

nchronas committed:
added boot counter

nchronas committed:
used a better name

nchronas committed:
added tool for epoch calculation

nchronas committed:
added fixes for boot counter.

nchronas committed:
added event logs in assert, a way to get a file name id

nchronas committed:
added ev states

nchronas committed:
added function definitions in time management

nchronas committed:
finished functionality in time management

nchronas committed:
fixed time management issues

nchronas committed:
WIP: Made custom uart rx routine, changed tx to dma

nchronas committed:
changed tc_tm pkt data into a pointer

nchronas committed:
refactored hdldc deframe

nchronas committed:
uart polling to IT changes

nchronas committed:
moved global obc_data struct in services

nchronas committed:
fixes for previous changes

nchronas committed:
updates for test project

nchronas committed:
fixes for hk service, added tc_tm pkt the data buffer

nchronas committed:
fixed hk crt pkt for TM not TC.

nchronas committed:
added boot elapsed time function

nchronas committed:
addition for wod logs in ms

nchronas committed:
added wod event buffer

nchronas committed:
corrected frame and tx

nchronas committed:
cleaned project

nchronas committed:
added WIP in readme
